AbletonMCP_Remote_Script/handlers/audio.py

```
@@ -0,0 +1,253 @@
+    try:
+        clip = _get_audio_clip(song, track_index, clip_index)
+
+        warp_mode_map = {
+            "beats": 0, "tones": 1, "texture": 2,
+            "re_pitch": 3, "complex": 4, "complex_pro": 5,
+        }
+        if warp_mode.lower() not in warp_mode_map:
+            raise ValueError(
+                "Invalid warp mode. Must be one of: beats, tones, texture, "
+                "re_pitch, complex, complex_pro"
+            )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)